frontend/src/views/TE/Upload.vue

```diff
@@ -44,75 +44,59 @@
         </div>
       </div>
       <v-divider v-if="getFiles().length"></v-divider>
-      <v-list two-line>
-        <template v-for="[file, status] in getFiles()">
-          <div :key="file.upload.uuid">
-            <v-list-item>
-              <v-list-item-content>
-                <v-list-item-subtitle
-                  :style="{ color: 'red' }"
-                  v-if="status.error"
-                  >{{ status.error }}
-                </v-list-item-subtitle>
-                <v-list-item-title>{{ file.name }}</v-list-item-title>
-              </v-list-item-content>
-              <v-list-item-action :style="{ flexDirection: 'row' }">
-                <v-btn v-if="status.error" @click="deleteFile(file)"
-                  >remove</v-btn
-                >
-                <v-icon color="green" v-else-if="status.complete"
-                  >fa-check-circle</v-icon
-                >
-                <div v-else>
-                  <v-progress-circular
-                    :value="status.progress"
-                    color="green"
-                    :style="{ marginRight: '16px' }"
-                  ></v-progress-circular>
-                  <v-btn @click="cancelFile(status)" v-if="status.uploading">
-                    Cancel
-                  </v-btn>
-                </div>
-              </v-list-item-action>
-            </v-list-item>
-            <v-divider></v-divider>
-          </div>
-        </template>
-      </v-list>
+      <upload-files
+        @delete-file="deleteFile"
+        @cancel-file="cancelFile(...arguments)"
+        @cancel-queued-file="cancelQueueFile"
+        :completedFiles="completedFiles"
+        :queuedFiles="queue"
+        :uploadingFiles="getFiles()"
+      >
+      </upload-files>
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import { Component, Mixins } from "vue-property-decorator";
+import { Component, Mixins, Watch } from "vue-property-decorator";
 import { mapState } from "vuex";
 import * as Spark from "spark-md5";
 import _ from "lodash";
 import VueDropzone from "vue2-dropzone";
 import "vue2-dropzone/dist/vue2Dropzone.min.css";
 import moment from "moment";
+import Promise from "bluebird";
 import firebase from "firebase/app";
 import "firebase/database";
 import "firebase/storage";
 import BaseTaskMixin from "@/components/TE/BaseTaskMixin";
+import UploadFileList from "@/components/TE/UploadFileList.vue";
+
+import { getTaskId, getProjectDomain } from "@/utility";
+import UploadFiles from "@/components/TE/UploadFilesList.vue";
```

Comment:
It is confusing to have two similar components: `UploadFilesList` and `UploadFileList`.

I am not sure it's worth extracting `UploadFilesList` into a separate component. Let's keep it within `Upload` component.